adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -430,8 +430,11 @@ public AuthenticationResult getTokenForCode(String code) throws IOException, Aut
     private AuthenticationResult postMessage(String requestMessage, Map<String, String> headers)
             throws IOException, AuthenticationException {
         AuthenticationResult result = null;
+        HttpEvent httpEvent = startHttpEvent();
```

Comment:
cannot be final, look at line 456 for challenge request header 

---
In reply to: [76139458](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/703#discussion_r76139458) [](ancestors = 76139458)